### ISSUE TITLE
cachemap should display caches at modified cords, not at original

### ISSUE THREAD

**Issue by kojoty** [opencaching/okapi#455]
HI @wrygiel,

We have issue [https://github.com/opencaching/opencaching-pl/issues/913](https://github.com/opencaching/opencaching-pl/issues/913).

Do you think it could be easy implemented at OKAPI side?

**Comment by wrygiel** [comment]
> Do you think it could be easy implemented at OKAPI side?

OKAPI is using a separate index of cache locations to make the tile generation faster. This index is global (one index per all users). Displaying caches at modified coords would either require N indexes instead of 1 (where N is the number of users), or dropping the index altogether. Only the second of these options is feasible (dropping the index), but I'm not sure how performance would suffer.

**Comment by wrygiel** [comment]
> I'm not sure how performance would suffer

Just to clarify, I'm sure that it **would** suffer, but I cannot tell how much. My tentative guess is that it would make map generation **at least** 10 times slower, especially for lower zoom levels.

I believe that we would not only the resigning from this particular custom index of mine - we also wouldn't be able to use the default index (the one in `caches` table). We would need to query the `services/caches/search/bbox` method with `location_source=alt_wpt:user-coords` *for every tile request*. And, as far as I know, this `location_source` doesn't use any index whatsoever.

**Comment by kojoty** [comment]
@wrygiel, thanks for answer
I think we can close this issue - it seems there is no good way to implement it.

**Comment by wrygiel** [comment]
> I think we can close this issue - it seems there is no good way to implement it.

I think I can see a way of solving this, but it certainly isn't easy :) and I won't have time to implement this any time soon...  We need to use two indexes - the existing one, plus a set of new ones (one small user-coords index per user). If the second index is cached in APC, this would work almost as fast as the current solution.

**Comment by andrixnet** [comment]
Could this be related to the ability to show/hide caches other then current on cachepage popup map and the ability to show waypoints of a cache on the map ? (see here https://github.com/opencaching/opencaching-pl/issues/90 )

**Comment by andrixnet** [comment]
Also, IMO such a feature should display both : the original cache coordinates (with proper icon), given that those coords may have significance for some field puzzle caches and not only. 
And the user coords with a different icon, also different from owner defined waypoints

**Comment by following5** [comment]
> Could this be related to the ability to show/hide caches other then current on cachepage popup map and the ability to show waypoints of a cache on the map ? (see here opencaching/opencaching-pl#90 )

I don't think it is related. Generating minimap for just one cache would be a very different thing than the current OKAPI map.

**Comment by following5** [comment]
This issue is on hold, because there are different requests:
* Show modified coords _instead of_ original coords (kojoty), or
* show modified coords separately, with special icon (andrixnet).

OCPL developers/users must decide what the map should display.

**Comment by kojoty** [comment]
My thoughts: 
- modified coords are useful for quizes, which needs to be resolved offline, sometimes long time before the trip to the place, where cache is hidden in terrain. 
- then user usually doesn't need to see cache on original coords - useful is only new "real" position
- I think, that probably it would be well if cache on modified coords should have some graphical identification (different icon - some addition to icon) - there is no reason to show original coords on map.

@andrixnet what do you think?
